apps/web/src/app/(signing)/sign/[token]/signature-field.tsx

```diff
@@ -167,8 +178,17 @@ export const SignatureField = ({ field, recipient }: SignatureFieldProps) => {
 
             <SignaturePad
               id="signature"
+              signature={{
+                value: localSignature?.value ?? '',
+                type: localSignature?.type ?? SignatureType.DRAW,
+              }}
               className="border-border mt-2 h-44 w-full rounded-md border"
-              onChange={(value) => setLocalSignature(value)}
+              onChange={(value: string | null, isUploaded: boolean) => {
+                setLocalSignature({
+                  value,
+                  type: isUploaded ? SignatureType.UPLOAD : SignatureType.DRAW,
+                });
+              }}
```

Comment:
for complex onChange function, it's better them to define them as function and use them in JSX